power_grid_model_c/power_grid_model/include/power_grid_model/main_model_impl.hpp

```diff
@@ -801,12 +801,8 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
 
     CalculationInfo calculation_info() const { return calculation_info_; }
 
-    void check_no_experimental_features_used(Options const& options) const {
-        if (options.calculation_type == CalculationType::state_estimation &&
-            state_.components.template size<GenericCurrentSensor>() > 0) {
-            throw ExperimentalFeature{"State estimation", "current sensors"};
-        }
-    }
+    // This function is intentionally left empty to maintain API compatibility.
+    void check_no_experimental_features_used(Options const& /*options*/) const {}
```

Comment:
```suggestion
        if (options.calculation_type == CalculationType::state_estimation &&
            options.calculation_method != CalculationMethod::iterative_linear &&
            state_.components.template size<GenericCurrentSensor>() > 0) {
            throw ExperimentalFeature{"State estimation", "current sensors"};
        }
```

Reply:
@TonyXiang8787 we need this, otherwise the sufficient observability check is no longer sufficient

Reply:
@petersalemink95 , @Jerry-Jinfeng-Guo and I agreed to throw an experimental feature.